src/httpServer/server.ts

```diff
@@ -61,18 +62,48 @@ export class DBOSHttpServer {
    * Register HTTP endpoints and attach to the app. Then start the server at the given port.
    * @param port
    */
-  listen(port: number) {
-    // Start the HTTP server.
+  async listen(port: number) {
+    try {
+      await this.checkPortAvailability(port, "127.0.0.1");
+      await this.checkPortAvailability(port, "::1");
+    } catch (error) {
+      this.logger.warn(`Port ${port} is already used. Please use the -p option to choose another port.`);
+      process.exit(1);
+    }
+
     const appServer = this.app.listen(port, () => {
```

Comment:
Is it possible to simply define an error handler for the `appServer`? Example https://github.com/koajs/koa/issues/1517

This way, we don't need to check port availability in a separate function.

```typescript
const appServer = this.app.listen(...)
appServer.on('error', httpErrorHandler)

function httpErrorHandler (err) {
// If it's EADDRINUSE, print port is already used, use `-p` to choose another port
}
```